js/src/framework.ts

```diff
@@ -210,10 +255,43 @@ export async function runEvaluator(
   if (typeof evaluator.data === "string") {
     throw new Error("Unimplemented: string data paths");
   }
-  const dataResult = evaluator.data();
-  let data = null;
+  let dataResult =
+    typeof evaluator.data === "function" ? evaluator.data() : evaluator.data;
+
+  if ("_type" in dataResult) {
+    if (dataResult._type !== "BaseExperiment") {
+      // For some reason, the typesystem won't let me check if dataResult._type === "BaseExperiment"
+      throw new Error("Invalid _type");
+    }
+    if (!experiment) {
+      throw new Error(
+        "Cannot use BaseExperiment() without connecting to Braintrust (you most likely set --no-send-logs)"
+      );
+    }
+    let name = dataResult.name;
+    if (isEmpty(name)) {
+      const baseExperiment = await experiment.fetchBaseExperiment();
+      if (!baseExperiment) {
+        throw new Error("BaseExperiment() failed to fetch base experiment");
+      }
+      name = baseExperiment.name;
+    }
+    dataResult = init(evaluator.projectName, {
+      experiment: name,
+      open: true,
+    }).asDataset();
+  }
+
+  let data: EvalCase<unknown, unknown>[] = [];
```

Comment:
unrelated nit on the line below: we should just be able to write `let dataResult = await (typeof ...)` on line 258 and remove the `if` here.

Reply:
will let you refactor this at some point :)